### PR TITLE
PHPCS: rename class file for gallery settings

### PR DESCRIPTION
**Pull request by bindlegirl**
- Adding renamed copy of class file
- Add changelog

Renaming functions.gallery.php to class-jetpack-gallery-settings.php.

#### Changes proposed in this Pull Request:

* Adding a new file
* Deleting the old one
* Changing files that mention or call the old file 

#### Jetpack product discussion

none
#### Does this pull request change what data or activity we track or use?

no
#### Testing instructions:

Test if galleries are working

### CODE REVIEW

**Comment by bindlegirl** [comment]
The D79680-code will need approval as well.
Once this PR is merged, D79708-code should also get approved.

**Review by jeherve — APPROVED** [review]

**Comment by kraftbj** [comment]
r245110-wpcom

**Pull request merged**
